crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -152,8 +153,17 @@ where
                     Render,
                     (
                         prepare_previous_view_projection_uniforms,
-                        batch_and_prepare_render_phase::<Opaque3dPrepass, MeshPipeline>,
-                        batch_and_prepare_render_phase::<AlphaMask3dPrepass, MeshPipeline>,
+                        (
+                            reserve_batch_buffer::<Opaque3dPrepass, MeshPipeline>,
+                            reserve_batch_buffer::<AlphaMask3dPrepass, MeshPipeline>,
+                        )
+                            .before(allocate_batch_buffer::<MeshPipeline>)
+                            .after(clear_batch_buffer::<MeshPipeline>),
+                        (
+                            batch_and_prepare_render_phase::<Opaque3dPrepass, MeshPipeline>,
+                            batch_and_prepare_render_phase::<AlphaMask3dPrepass, MeshPipeline>,
+                        )
+                            .after(allocate_batch_buffer::<MeshPipeline>),
```

Comment:
This starts to look like we maybe want a batching plugin that is generic over render phase item and the base specialisation pipeline or something.

Reply:
Agreed there. This is getting unwieldy to manage.